include/llbuild/Core/BuildDB.h

```diff
@@ -31,10 +31,10 @@ class BuildDB {
   virtual ~BuildDB();
 
   /// Get the current build iteration.
-  virtual uint64_t getCurrentIteration() = 0;
+  virtual uint64_t getCurrentIteration(std::string* error_out) = 0;
```

Comment:
What does this return on error? Is it ok to use 0 as an error code here or should we promote this to be an argument?